utils/release_tracking.py

```diff
@@ -0,0 +1,99 @@
+    future_repo_dir = os.path.join(clone_dir, repo_name)
+    if not os.path.isdir(future_repo_dir):
+        command = f"git clone {repo_link} -b {branch} {future_repo_dir}"
+        subprocess.check_output(command, shell=True)
+        command = f"git submodule update --init"
+        return subprocess.check_output(command, shell=True, cwd=future_repo_dir)
+
+
+def git_toml_deps(toml_path, updatable_deps, deps_repos):
+    lines = None
+    with open(toml_path, 'r') as f:
+        lines = f.readlines()
+
+
+    output_lines = []
+    for line in lines:
+        for dep in updatable_deps.keys():
+            if line.startswith(dep + " ="):
+                # All occurences
+                line = line.replace("'", '"')
+                if "{" not in line:
+                    # First occurence
+                    line = line.replace('"', '{ version = "', 1)
+                    line = line.rstrip() + ' }\n'
+
+                if 'path' not in line:
+                    line = re.sub(r'version = "[0-9\.]+"', f'path = "{deps_repos[dep]}"', line)
+        output_lines.append(line)
+
+    with open(toml_path, 'w') as f:
+        f.writelines(output_lines)
+    git_cmd = 'git diff'
+    print(subprocess.check_output(git_cmd, shell=True, cwd=os.path.dirname(toml_path)).decode('utf-8'))
+
+
+def main(argv=[], prog_name=''):
+    parser = argparse.ArgumentParser(prog='ReleaseTracker',
+                                     description='Modifies the parsec Cargo.toml files to use the '
+                                                 'main branches of parallaxsecond dependencies in '
+                                                 'preparation for their publishing and release')
+    parser.add_argument('--clone_dir',
+                        required=True,
+                        help='Existing directory into which repositories should be cloned')
+    parser.add_argument('paths', nargs='+', help='Paths to Cargo.toml files to be modified')
+    args = parser.parse_args()
+
+    # The order is important!
+    parallaxsecond_deps = {
+        'psa-crypto-sys': 'rust-psa-crypto',
+        'psa-crypto': 'rust-psa-crypto',
+        'tss-esapi-sys': 'rust-tss-esapi',
+        'tss-esapi': 'rust-tss-esapi',
+        'cryptoki-sys': 'rust-cryptoki',
+        'cryptoki': 'rust-cryptoki',
+        'parsec-interface': 'parsec-interface-rs',
+        'parsec-client': 'parsec-client-rust',
+    }
+
+    repo_paths = { repo_name: f'{args.clone_dir}/{repo_folder}/{repo_name}' \
+                   for repo_name, repo_folder in parallaxsecond_deps.items() }
+    repo_paths['parsec-interface'] = f'{args.clone_dir}/parsec-interface-rs'
+    repo_paths['parsec-client'] = f'{args.clone_dir}/parsec-client-rust'
+
+    repo_branches = { repo_name: 'main' for repo_name in parallaxsecond_deps.keys() }
+    repo_branches['tss-esapi-sys'] = '7.x.y'
+    repo_branches['tss-esapi'] = '7.x.y'
+
+    for repo_name, repo_folder in parallaxsecond_deps.items():
```

Comment:
We could instead use [patch](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section) in the toml file. 

```
[patch.crates-io]
psa-crypto = { git = 'https://github.com/parallaxsecond/rust-psa-crypto.git', branch = "main" }
```

Reply:
That is a very nice feature that I didn't know about.  It would remove the need for the `git_toml_deps` method?

Reply:
For future reference:

Adding the `cargo update` step here was crucial to make it work as otherwise we would be using the Cargo.lock file version instead of the patch one